src/extension/src/TelemetryWriter.py

```diff
@@ -96,6 +98,7 @@ def write_event(self, message, event_level=Constants.TelemetryEventLevel.Informa
 
             self.__delete_older_events()
 
+            task_name = self.__task_name if task_name == Constants.TELEMETRY_TASK_NAME else task_name
```

Comment:
Shouldn't this be:
task_name = self.__task_name if task_name == Constants.TELEMETRY_TASK_NAME else task_name **+ self.__task_name_watermark**